crates/swift-bridge-ir/src/codegen/generate_c_header.rs

```diff
@@ -141,24 +143,61 @@ typedef struct {option_ffi_name} {{ bool is_some; {ffi_name} val; }} {option_ffi
-                            variants += &v;
+                            let variant = format!("{}${}, ", ffi_name, variant.name);
+                            variants += &variant;
                         }
 
                         let maybe_vec_support = if ty_enum.has_one_or_more_variants_with_data() {
                             "".to_string()
                         } else {
                             vec_transparent_enum_c_support(&ty_enum.swift_name_string())
                         };
-
+                        let mut variant_fields = "".to_string();
+                        for variant in ty_enum.variants.iter() {
+                            match &variant.fields {
+                                StructFields::Named(_) => {
+                                    todo!();
+                                }
+                                StructFields::Unnamed(unnamed_fields) => {
+                                    let mut params = vec![];
+                                    for unnamed_field in unnamed_fields.iter() {
+                                        let variant_field = BridgedType::new_with_type(
+                                            &unnamed_field.ty,
+                                            &self.types,
+                                        )
+                                        .unwrap();
+                                        let variant_field = variant_field.to_c();
+                                        params.push(format!(
+                                            "{} _{};",
+                                            variant_field, unnamed_field.idx
+                                        ));
+                                    }
+                                    let params = params.join(" ");
+                                    let variant_field = format!("typedef struct {ffi_name}$FieldOf{variant_name} {{{params}}} {ffi_name}$FieldOf{variant_name};", ffi_name = ffi_name, variant_name = variant.name, params = params);
+                                    variant_fields += &variant_field;
+                                    variant_fields += "\n";
+                                }
+                                StructFields::Unit => {
+                                    let variant_field = format!("typedef struct {ffi_name}$FieldOf{variant_name} {{uint8_t _private;}} {ffi_name}$FieldOf{variant_name};", ffi_name = ffi_name, variant_name = variant.name);
+                                    variant_fields += &variant_field;
+                                    variant_fields += "\n";
+                                }
+                            }
+                        }
                         let enum_decl = format!(
-                            r#"typedef enum {ffi_tag_name} {{ {variants}}} {ffi_tag_name};
-typedef struct {ffi_name} {{ {ffi_tag_name} tag; }} {ffi_name};
+                            r#"#include <stdint.h>
```

Comment:
We don't need this unconditional `stdint.h`.

Instead we can insert includes as needed like we do for structs https://github.com/chinedufn/swift-bridge/blob/0f436bc8f55f68cf7d023e5e04e7898c428776bb/crates/swift-bridge-ir/src/codegen/generate_c_header.rs#L83-L89